src/services/pncWebSocketService.ts

```diff
@@ -0,0 +1,54 @@
+import { pncWebSocketClient } from 'services/pncWebSocketClient';
+import { uiLogger } from 'services/uiLogger';
+
+type TWebSocketListener = (messageEvent: MessageEvent) => void;
+const messageListeners: Array<TWebSocketListener> = [];
+
+const LISTENERS_WARNING_COUNT = 20;
```

Comment:
maybe a little bit arbitrary number

Reply:
We can update it accordingly later once we have better data available.